### PR TITLE
feat: expose Qute server settings

### PR DESCRIPTION
**Pull request by fbricon**
Part of https://github.com/redhat-developer/intellij-quarkus/issues/794

Screenshot 2023-08-29 at 13 30 29
Screenshot 2023-08-29 at 13 30 43

Signed-off-by: Fred Bricon <fbricon@gmail.com>

### CODE REVIEW

File: src/main/java/com/redhat/devtools/intellij/qute/settings/ProblemSeverity.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ *******************************************************************************/
+package com.redhat.devtools.intellij.qute.settings;
+
+import com.intellij.lang.annotation.HighlightSeverity;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Problem severity levels used by Qute LS
+ */
+//TODO move to lsp4ij?
```
[the remark this answers is not in the document]
Comment (fbricon) [reply src/main/java/com/redhat/devtools/intellij/qute/settings/ProblemSeverity.java:22]: yeah but you could have 4 problem severities in one LS, 3 in another, it makes mapping difficult

[the remark this answers is not in the document]
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/qute/settings/ProblemSeverity.java:22]: I need to read more the code but in final lsp provide diagnosticseverity which provide known fixed severities, no?

[the remark this answers is not in the document]
Comment (fbricon) [reply src/main/java/com/redhat/devtools/intellij/qute/settings/ProblemSeverity.java]: Using DiagnosticSeverity now

File: src/main/java/com/redhat/devtools/intellij/qute/lsp/QuteLanguageClient.java [hunk 2]
```diff
@@ -91,6 +106,28 @@ private void notifyQuteDataModelChanged(Set<String> uris) {
         }
     }
 
+    @Override
+    protected Object createSettings() {
+        return UserDefinedQuteSettings.getInstance(getProject()).toSettingsForQuteLS();
+    }
+
+    @Override
+    public void profileChanged(@NotNull InspectionProfile profile) {
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/qute/lsp/QuteLanguageClient.java:117]: @fbricon what do you think about that?

[the remark this answers is not in the document]
Comment (fbricon) [reply src/main/java/com/redhat/devtools/intellij/qute/lsp/QuteLanguageClient.java:117]: I don't want to make all clients unnecessarily register as profile listener if it's not needed. I'd prefer keeping ad-hoc listener when needed

**Pull request merged**